packages/experimental/Shadow/src/Shadow.types.ts

```diff
@@ -0,0 +1,22 @@
+import { IBorderTokens } from '@fluentui-react-native/tokens';
+import { ViewStyle, ViewProps, ColorValue } from 'react-native';
+
+export type ShadowDepth = '2' | '4' | '8' | '16' | '28' | '64';
+
+export interface ShadowTokens extends IBorderTokens {
+  ambient?: ViewStyle;
+  key?: ViewStyle;
+  color?: ColorValue;
+  backgroundColor?: ColorValue;
+  innerShadowSlot?: React.ComponentType<ViewProps>;
+  depth2?: ShadowTokens;
```

Comment:
I'm curious why the depths and the other tokens are part of the same interface. The current structure allows for recursion/nesting so you can do something like depth2.depth4.backgroundColor = whatever, which seems odd to me.

Reply:
yea we apply Tokens in layers which is kind of convenient but in this case it's also a little strange.  We'll flatten the nested Token objects based on the function passed to applyTokenLayers

`(layer) => layer === 'depth' + depth`

It that returns true, then the nested tokens will be merged on top of the underlying layer.  It's basically how the depth prop functions right now... but maybe that isn't the right approach for this component.  Let me think about it a little more.